src/useradd.c

```diff
@@ -385,6 +385,7 @@ static void get_defaults (void)
 			} else {
 				def_group = grp->gr_gid;
 				def_gname = xstrdup (grp->gr_name);
+				gr_free(grp);
```

Comment:
This should be a separate PR.

I'm not sure this is safe.  For instance, in the case where grp has the result of fgetgrent(), putgrent() should be used to free it.

Given that this is not a library, and will exit and free all its resources, I think it's safer to leave it be.  We've had patches not too  long ago which tried to free memory right before exiting, which caused crashes in some uncommon situations.

Reply:
Thank you for your reply. I updated the commit message Do you think I've got enough reasons in my commit message?
And have you checked the other commit?

Reply:
No, you are wrong here:

```
and the memory is malloced
through the prefix_getgr_nam_gid function ->
__gr_dup function.
```

It can also be getgrgid(), whose manpage says not to free.

This is not a long running program.  The dangers of an accidental double free or similar problem far outweigh the problems of a tiny bit of memory not being freed.

Reply:
OK, that makes sense